### PR TITLE
[docs] First pass at using Expo CLI docs page

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

Help people understand that they can use Expo CLI in any React Native project, and why they might want to do so :)

# How

Take things I know in my head and put them into markdown.

# Test Plan

Read the docs.

# Checklist

- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

### CODE REVIEW

File: docs/pages/bare/using-expo-cli.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import { Terminal } from '~/ui/components/Snippet';
+
+To use the local Expo CLI in a project created with `npx react-native init`, you will first need to install the `expo` package. This package includes the Expo Modules API and Expo CLI. For a complete explanation of how to install this, see the ["Install Expo modules" guide](/bare/installing-expo-modules.md). For most apps, it's as simple as running the following command in your project directory:
+
+<Terminal cmd={['npx install-expo-modules@latest']} />
+
+## Compiling and running your app
+
+Once you have the `expo` package installed, you can use the `npx expo run:ios` and `npx expo run:android` commands as alternatives to `npx react-native run-ios` and `npx react-native run-android`. You can pass in the `--device` flag to select a device/simulator to build for — this also works for any connected iOS device.
```
Comment (EvanBacon) [inline docs/pages/bare/using-expo-cli.mdx:14]: Maybe add info about how the Expo run commands (ios) add additional contextual logging and enable you to build onto a connected device. Unlike the community CLI, you can do this without global side-effects, or opening xcode to manually provision the build.

File: docs/pages/bare/using-expo-cli.mdx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- View native logs directly in the terminal.
+- [Automatic TypeScript support](/guides/typescript.md).
+- [Automatic web support](/guides/customizing-metro.md#adding-web-support-to-metro).
+- Integrates with [expo-dev-client](/develop/development-builds/introduction.md).
+- Improved native build log formatting by using Expo's own `xcpretty`-style tool that is built specifically for React Native apps.
+- It runs `pod install` automatically when needed.
+- `npx expo install` picks known compatible versions of dependencies for well-known packages.
+- `npx expo run:[android|ios]` and `npx expo start` automatically detect if another app is running on the default port and suggests using a different one.
+- Built-in support for serving your app over an [ngrok tunnel](/develop/development-builds/development-workflows/#tunnel-urls), which is useful for testing on physical devices on restricted networks.
+- Support for [expo-router](https://expo.github.io/router/docs).
```
Comment (EvanBacon) [inline docs/pages/bare/using-expo-cli.mdx:35]:
```suggestion
- Support for File-based routing with [`expo-router`](https://expo.github.io/router/docs). This automatically enables deep linking for every screen in your app.
```

File: docs/pages/bare/using-expo-cli.mdx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+## Starting the bundler
+
+`npx expo run:[android|ios]` will start the bundler/development server. You can skip that by passing in `--no-bundler` flag. Start the development server independently with `npx expo start --dev-client`.
+
+## Why we think you should use it instead of `npx react-native`
+
+Expo CLI commands provide several benefits over the similar commands in `@react-native-community/cli`, including:
+
+- Open up the Hermes debugger in one keystroke: <kbd>j</kbd>.
+- [Prebuild](/workflow/prebuild.md) makes upgrading your native projects to new React Native versions easier, among other benefits.
```
Comment (EvanBacon) [inline docs/pages/bare/using-expo-cli.mdx]:
```suggestion
- Supports [Continuous Native Generation](/workflow/prebuild.md) (CNG) for easy upgrades, white-labeling, simple third-party package setup.
```

Maybe we can slip in this new acronym

File: docs/pages/bare/using-expo-cli.mdx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+## Starting the bundler
+
+`npx expo run:[android|ios]` will start the bundler/development server. You can skip that by passing in `--no-bundler` flag. Start the development server independently with `npx expo start --dev-client`.
+
+## Why we think you should use it instead of `npx react-native`
+
+Expo CLI commands provide several benefits over the similar commands in `@react-native-community/cli`, including:
+
+- Open up the Hermes debugger in one keystroke: <kbd>j</kbd>.
```
Comment (EvanBacon) [inline docs/pages/bare/using-expo-cli.mdx:24]: Built-in monorepo support

Comment (EvanBacon) [reply docs/pages/bare/using-expo-cli.mdx:24]: install command / versions endpoint is another good feature.

Comment (Kudo) [reply docs/pages/bare/using-expo-cli.mdx:24]: Built-in react-devtools support (available on expo@latest). though without expo-go or expo-dev-client, user needs to manually reload their app to make the app showing in react-devtools.

**Comment by amandeepmittal** [comment]
There are some formatting nits for consistency reasons that I'd like to update. I'll create a follow up PR for this.

**Pull request merged**